### DISCUSSION TITLE
[Craft 4] Add .one() to Collections... or revive .first() for element queries 😬

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#11134] [Ideas] [answered]
Having eager-loaded elements returned as Collections in Craft 4 is awesome, as it greatly reduces the need for boilerplate when having to write code that loops over elements that may or may not be eager-loaded.

Unfortunately though, fetching a _single_ element in a lazy/eager agnostic way is still a little bit awkward, since there's no `.one()` method available to Collections. AFAIK, this is currently the best way to do it:

```twig
{% set image = entry.image.collect().first() %}
```

Which... works, but is arguably not as nice as the familiar `entry.image.one()`.

To increase API parity between the two interfaces and make it even easier to write code that works for both lazy and eager loaded elements, it'd be great if Craft could provide a `.one()` method for Collections.

**Comment by brandonkelly** [comment, marked as answer]
Good idea. Added a `one()` alias, for Craft 4.1! (6e56bda4ba04e23168ea8f738d5af27d4cdc0c43)

**Reply by mmikkel** [reply]
Awesome! Thanks a lot @brandonkelly 🙌

**Reply by brandonkelly** [reply]
Craft 4.1 is out with this!
